server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -532,15 +533,12 @@ public void enqueueManagerMessage(ManagerMessage m) {
     managerMessages.addLast(m);
   }
 
-  void acquireRecoveryMemory(KeyExtent extent) {
-    if (!extent.isMeta()) {
+  AutoCloseable acquireRecoveryMemory(TabletMetadata tabletMetadata) {
```

Comment:
This may not make much of a difference so could also ignore this comment but I'm wondering if we should just cache/re-use the AutoCloseable objects to avoid extra GC.

 I was thinking something like the following as TabletServer class level:

```java
  private static final AutoCloseable NOOP_CLOSEABLE = () -> {};
  private final AutoCloseable recoveryLockCloseable = recoveryLock::unlock;
```

Then the require method could look like:
```java
  AutoCloseable acquireRecoveryMemory(TabletMetadata tabletMetadata) {
    if (tabletMetadata.getExtent().isMeta() || tabletMetadata.getLogs().isEmpty()) {
      return NOOP_CLOSEABLE;
    } else {
      recoveryLock.lock();
      return recoveryLockCloseable;
    }
  }
```

Reply:
I partially implemented this in 0a341af, only added NOOP_CLOSEABLE.   Thinking the case w/o walogs is the most frequent.  The case w/ walogs is less frequent and will already take a lot longer because of recovery so thought that less code was better for that case over saving an object allocation.